datanode/networkhistory/store/store.go

```diff
@@ -877,14 +877,22 @@ func createNode(ctx context.Context, log *logging.Logger, repoPath string, maxMe
 // The LibP2P Resource manager protects the IPFS node from malicious and non-malicious attacks, the limits used to enforce
 // these protections are based on the max memory and max file descriptor limits set in the swarms resource manager config.
 // This method overrides the defaults and sets limits that we consider sensible in the context of a data-node.
-func setLibP2PResourceManagerLimits(repo repo.Repo, maxMemoryPercent uint8) error {
+func setLibP2PResourceManagerLimits(log *logging.Logger, repo repo.Repo, maxMemoryPercent uint8) error {
 	cfg, err := repo.Config()
 	if err != nil {
 		return fmt.Errorf("failed to get repo config:%w", err)
 	}
 
+	totalMem := memory.TotalMemory()
+	// The above call can return 0 if the underlying system call errors which can lead to further panics,
+	// check for 0 and set to default to protect against this.
+	if totalMem == 0 {
+		totalMem = 100_000_000
+		log.Warningf("failed to determine total memory from system call, setting to %d bytes", totalMem)
+	}
+
 	// Set the maximum to a quarter of the data-nodes max memory
-	maxMemoryString := humanize.Bytes(uint64(float64(memory.TotalMemory()) * (float64(maxMemoryPercent) / (4 * 100))))
```

Comment:
Theres other places we call `memory.TotalMemory()`. We should handle `0` there too:
```
// only set max memory if user didn't require 100%
if memFactor != 1 {
	totalMem := memory.TotalMemory()
	debug.SetMemoryLimit(int64(float64(totalMem) * memFactor))
}
```